executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -35,7 +36,7 @@ pub fn split_out_machines<'a, T: FieldElement>(
     let mut machines: Vec<Box<dyn Machine<T>>> = vec![];
 
     let all_witnesses = witness_cols.iter().map(|c| &c.poly).collect::<HashSet<_>>();
-    let mut remaining_witnesses = all_witnesses.clone();
+    let mut remaining_witnesses: HashSet<&'a PolynomialReference> = all_witnesses.clone();
```

Comment:
Why did you have to specify the full type here?

If really needed, I would have done it in the previous line, either at `collect::<...>` or at `let all_witnesses: ...`, but not both.

Reply:
The `'a` was the thing I mainly needed.

Reply:
Still could be done only in the previous line.

Reply:
Ah, now I see! Here it is: https://github.com/powdr-org/powdr/pull/359